mentat/code_context.py

```diff
@@ -262,62 +206,26 @@ async def _get_code_message(
         code_message += ["Code Files:\n"]
         meta_tokens = count_tokens("\n".join(code_message), model)
 
-        # Add user-included features
-        included_features = self._get_include_features()
-        included_feature_tokens = sum(
-            await count_feature_tokens(included_features, model)
-        )
-        remaining_tokens = max(0, max_tokens - meta_tokens - included_feature_tokens)
         auto_tokens = config.auto_tokens
-        if remaining_tokens == 0 or config.auto_tokens == 0:
-            self.features = sorted(
-                included_features, key=lambda f: f.path.relative_to(git_root)
-            )
+        remaining_tokens = max(0, min(auto_tokens, max_tokens - meta_tokens))
+
+        if remaining_tokens == 0:
```

Comment:
I was disappointed this check was necessary. Logically the function should be the same if you always call `_get_all_features` and then you don't even need to compute `remaining_tokens`. But it's unfortunately pretty slow.

Reply:
Oh sorry I didn't mean to change the behavior. I feel like a test should have caught that. I think the behavior is better this way because I don't think users should think about their `include_token` budget and `auto_token` budget. They should just think about their token budget and it should be up to us how to best use it.